templates/types/streaming/fastapi/app/api/routers/models.py

```diff
@@ -124,6 +126,12 @@ def from_source_node(cls, source_node: NodeWithScore):
         metadata = source_node.node.metadata
         url = metadata.get("URL")
 
+        # if metadata has pipeline_id, get file url from LLamaCloudFileService
+        pipeline_id = metadata.get("pipeline_id")
+        if pipeline_id:
+            file_name = metadata.get("file_name")
+            url = LLamaCloudFileService.get_file_url(file_name, pipeline_id)
```

Comment:
This fetches data from the llama cloud service, which means it can have network errors. How about we add error handling here that accepts the error and uses the null URL?

Reply:
good idea